adi/cn0566.py

```diff
@@ -0,0 +1,620 @@
+        self.v0_vdd1v8_scale = 1.0 + (10.0 / 10.0)  # Resistances in k ohms.
+        self.v1_vdd3v0_scale = 1.0 + (10.0 / 10.0)
+        self.v2_vdd3v3_scale = 1.0 + (10.0 / 10.0)
+        self.v3_vdd4v5_scale = 1.0 + (30.1 / 10.0)
+        self.v4_vdd_amp_scale = 1.0 + (69.8 / 10.0)
+        self.v5_vinput_scale = 1.0 + (30.1 / 10.0)
+        self.v6_imon_scale = 1.0  # LTC4217 IMON = 50uA/A * 20k = 1 V / A
+        self.v7_vtune_scale = 1.0 + (69.8 / 10.0)
+
+        ### Initialize ADF4159 / Local Oscillator ###
+
+        self.lo = 10.5e9
+
+        BW = 500e6 / 4
+        num_steps = 1000
+        self.freq_dev_range = int(
+            BW
+        )  # frequency deviation range in Hz.  This is the total freq deviation of the complete freq ramp
+        self.freq_dev_step = int(
+            BW / num_steps
+        )  # frequency deviation step in Hz.  This is fDEV, in Hz.  Can be positive or negative
+        self.freq_dev_time = int(
+            1e3
+        )  # total time (in us) of the complete frequency ramp
+        self.ramp_mode = "disabled"  # ramp_mode can be:  "disabled", "continuous_sawtooth", "continuous_triangular", "single_sawtooth_burst", "single_ramp_burst"
+        self.delay_word = 4095  # 12 bit delay word.  4095*PFD = 40.95 us.  For sawtooth ramps, this is also the length of the Ramp_complete signal
+        self.delay_clk = "PFD"  # can be 'PFD' or 'PFD*CLK1'
+        self.delay_start_en = 0  # delay start
+        self.ramp_delay_en = 0  # delay between ramps.
+        self.trig_delay_en = 0  # triangle delay
+        self.sing_ful_tri = 0  # full triangle enable/disable -- this is used with the single_ramp_burst mode
+        self.tx_trig_en = 0  # start a ramp with TXdata
+        # pll.clk1_value = 100
+        # pll.phase_value = 3
+        self.powerdown = 0
+        self.enable = 0  # 0 = PLL enable.  Write this last to update all the registers
+
+        ### Initialize GPIOS / set outputs ###
+        self.gpios.gpio_vctrl_1 = 1  # Onboard PLL/LO source
+        self.gpios.gpio_vctrl_2 = 1  # Send LO to TX circuitry
+
+        self.gpios.gpio_div_mr = 0  # TX switch toggler divider reset
+        self.gpios.gpio_div_s0 = 0  # TX toggle divider lsb (1s)
+        self.gpios.gpio_div_s1 = 0  # TX toggle divider 2s
+        self.gpios.gpio_div_s2 = 0  # TX toggle divider 4s
+        self.gpios.gpio_rx_load = 0  # ADAR1000 RX load (cycle through RAM table)
+        self.gpios.gpio_tr = 0  # ADAR1000 transmit / receive mode. RX = 0 (assuming)
+        self.gpios.gpio_tx_sw = (
+            0  # Direct control of TX switch when div=[000]. 0 = TX_OUT_2, 1 = TX_OUT_1
+        )
+        # Read input
+        self.muxout = (
+            self.gpios.gpio_muxout
+        )  # PLL MUXOUT, assign to PLL lock in the future
+
+    def set_tx_sw_div(self, div_ratio):
+        """ Set TX switch toggle divide ratio. """
+        state_map = {0: 0, 2: 1, 4: 2, 8: 3, 16: 4, 32: 5, 64: 6, 128: 7}
```

Comment:
Need more doc for possible options of div_ratio

Reply:
Done and tested.